Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksProducerFilter.cc

```diff
@@ -0,0 +1,211 @@
+  triggerEvent_                       = iConfig.getParameter<edm::InputTag>("TriggerEventLabel");
+  theTriggerResultsLabel_             = iConfig.getParameter<edm::InputTag>("TriggerResultLabel");
+
+  // define tokens for access
+  tok_trigEvt_  = consumes<trigger::TriggerEvent>(triggerEvent_);
+  tok_trigRes_  = consumes<edm::TriggerResults>(theTriggerResultsLabel_);
+
+  edm::LogInfo("HcalIsoTrack") << "Use process name " << processName_
+			       << " Labels " << triggerEvent_ << " and "
+			       << theTriggerResultsLabel_ << " selecting "
+			       << trigNames_.size() << " triggers\n";
+#ifdef EDM_ML_DEBUG
+  std::cout << "HcalIsoTrack: Use process name " << processName_
+	    << " Labels " << triggerEvent_ << " and "
+	    << theTriggerResultsLabel_ << " selecting "
+	    << trigNames_.size() << " triggers" << std::endl;
+#endif
+  for (unsigned int k=0; k<trigNames_.size(); ++k) {
+    edm::LogInfo("HcalIsoTrack") << "Trigger[" << k << "] " << trigNames_[k]
+				 << std::endl;
+#ifdef EDM_ML_DEBUG
+    std::cout << "HcalIsoTrack: Trigger[" << k << "] " << trigNames_[k]
+	      << std::endl;
+#endif
+  }
+}
+
+AlCaIsoTracksProducerFilter::~AlCaIsoTracksProducerFilter() {}
+
+bool AlCaIsoTracksProducerFilter::filter(edm::Event& iEvent,
+					 edm::EventSetup const& iSetup) {
+  ++nAll_;
+  edm::LogInfo("HcalIsoTrack") << "Run " << iEvent.id().run() << " Event " 
+			       << iEvent.id().event() << " Luminosity " 
+			       << iEvent.luminosityBlock() << " Bunch " 
+			       << iEvent.bunchCrossing() << std::endl;
+#ifdef EDM_ML_DEBUG
+  std::cout << "HcalIsoTrack: Run " << iEvent.id().run() << " Event " 
+	    << iEvent.id().event() << " Luminosity " 
+	    << iEvent.luminosityBlock() << " Bunch " 
+	    << iEvent.bunchCrossing() << std::endl;
+#endif
+  
+  //Find if the event passes one of the chosen triggers
+  bool triggerSatisfied(false);
+  if (trigNames_.size() == 0) {
+    triggerSatisfied = true;
+  } else {
+    trigger::TriggerEvent triggerEvent;
+    edm::Handle<trigger::TriggerEvent> triggerEventHandle;
+    iEvent.getByToken(tok_trigEvt_, triggerEventHandle);
+    if (!triggerEventHandle.isValid()) {
+      edm::LogWarning("HcalIsoTrack") << "Error! Can't get the product "
+				      << triggerEvent_.label() ;
+    } else {
+      triggerEvent = *(triggerEventHandle.product());
+
+      /////////////////////////////TriggerResults
+      edm::Handle<edm::TriggerResults> triggerResults;
+      iEvent.getByToken(tok_trigRes_, triggerResults);
+      if (triggerResults.isValid()) {
+	std::vector<std::string> modules;
+	const edm::TriggerNames & triggerNames = iEvent.triggerNames(*triggerResults);
+	const std::vector<std::string> & triggerNames_ = triggerNames.triggerNames();
+	for (unsigned int iHLT=0; iHLT<triggerResults->size(); iHLT++) {
+	  int hlt    = triggerResults->accept(iHLT);
+	  for (unsigned int i=0; i<trigNames_.size(); ++i) {
```

Comment:
it seems that if hlt==0, there is no reason to execute this for loop. Is that correct?